Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
Short term, yes.
Long term, I'd like to stop `getErrors()` and use just `analysis.errors` notifications.
This implementation is more general purpose IMHO - sending updates as we get them.